### DISCUSSION TITLE
Usage de la nouvelle fonction enveloppe

### DISCUSSION THREAD

**Discussion by nmathey** [MadeInPierre/finalynx#86] [Q&A] [answered]
Hello,
 
Au top la #70 ;)
Par contre je galère à l'utiliser :D 

Par exemple si je veux juste récupérer. l'entièrement de mon CTO chez IB nommé "CTO - IB" sous Finary et bien fetché par l'API je dois (?):

`cto_ib = Envelope("CTO IB", "CTO - IB", date(2021,5,1), key="CTO - IB")`

puis dans un de mes Folder (en l'occurrence Long terme > Stocks > World)

```
Line(
        "Stock World",
        envelope=cto_ib,
        target=TargetRatio(0),
         ),
```

Mais au résultat
`└── ↯    0 € [CTO - IB] Stock World 0% → 0%      +376 €  `

Alors que sur le rendu du l'API c'est > 0 avec tous les bons détails.

**Comment by MadeInPierre** [comment, marked as answer]
Salut !

C'est sûr qu'il manque des tutos, pas facile de deviner comme ça, il faudrait que je me motive mais c'est pas gagné :upside_down_face: 

J'ai l'impression que tu souhaites utiliser la fonction qui remplit un dossier automatiquement avec le contenu d'une enveloppe entière dans Finary. Dans ce cas, seuls les `Folder` peuvent créer des lignes mais tu as utilisé une `Line` dans ton exemple. Du coup je verrais plutôt cette structure : 

```python
cto_ib = Envelope("CTO IB", "CTO - IB", date(2021,5,1), key="CTO - IB")

portfolio = Portfolio(children=[
    Folder(  # <-- seuls les folders peuvent s'auto-remplir
        "Stock World",
        envelope=cto_ib,
        target=TargetRatio(0),
        # <-- pas besoin de mettre children=[...], le Folder va créer les lignes
     ),  
]
```

Si l'objectif était de juste préciser à quelle enveloppe appartient cette ligne (soit pour du simple affichage, soit pour différentier deux lignes avec le même id), alors la structure serait celle-ci : 

```python
cto_ib = Envelope("CTO IB", "CTO - IB", date(2021,5,1), key="CTO - IB")

portfolio = Portfolio(children=[
    Line(
        "Stock World",
        key="FINARY_ID" # <-- nécessaire si "Stock World" n'est pas le même nom sur Finary ou non unique
        envelope=cto_ib,
        target=TargetRatio(0),
     ),
]
```

Bonus, pour plus d'infos sur comment déclarer une enveloppe, [voir la page dédiée](https://finalynx.readthedocs.io/en/latest/apidocs/finalynx/finalynx.portfolio.envelope.html#finalynx.portfolio.envelope.Envelope)

**Reply by nmathey** [reply]
C'est parfait en décrivant l'enveloppe directement sous 'Folder' c'est exactement ce que je recherchais!
